expo push notifications guide and example. (#18952)
supabase/supabase: apps/docs/pages/guides/functions/examples/push-notifications.mdx

  body: string
}

interface WebhookPayload {
  type: 'INSERT' | 'UPDATE' | 'DELETE'
  table: string
  record: Notification
  schema: 'public'
  old_record: null | Notification
}

const supabase = createClient(
  Deno.env.get('SUPABASE_URL')!,
  Deno.env.get('SUPABASE_SERVICE_ROLE_KEY')!
)

Deno.serve(async (req) => {
  const payload: WebhookPayload = await req.json()
  const { data } = await supabase
    .from('profiles')
    .select('expo_push_token')
    .eq('id', payload.record.user_id)
    .single()

  const res = await fetch('https://exp.host/--/api/v2/push/send', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${Deno.env.get('EXPO_ACCESS_TOKEN')}`,
    },
    body: JSON.stringify({
      to: data?.expo_push_token,
      sound: 'default',
      body: payload.record.body,
    }),
  }).then((res) => res.json())

  return new Response(JSON.stringify(res), {
    headers: { 'Content-Type': 'application/json' },
  })
})
```

## Create the database webhook

Navigate to the [Database Webhooks settings](https://supabase.com/dashboard/project/_/database/hooks) in your Supabase Dashboard.

1. Enable and create a new hook.

> 1. Conditions to fire webhook:
>    1.1. Table: `notifications`
>    1.1. Events: `Insert`

Author: ```suggestion
1. Conditions to fire webhook: Select the `notifications` table and tick the `Insert` event. 
```

1. Webhook configuration: Supabase Edge Functions.

> 1. Edge Function:
>    1.1. Method: `POST`
>    1.1. Select which edge function to trigger: `push`
>    1.1. Timeout: `1000`
> 1. HTTP Headers:
>    1.1. Content-type: `application/json`
>    1.1. Click "Add new header" > "Add auth header with service key.
> 1. Click "Create webhook".

Author: ```suggestion
1. Edge Function: Select the `push` edge function and leave the method as `POST` and timeout as `1000`. 
1. HTTP Headers: Click "Add new header" > "Add auth header with service key and leave Content-type: `application/json`.
1. Click "Create webhook".
```
